sympy/functions/special/delta_functions.py

```diff
@@ -175,6 +175,41 @@ def is_simple(self, x):
             return p.degree() == 1
         return False
 
+    def _eval_rewrite_as_Piecewise(self, *args):
+        """Represents DiracDelta in a Piecewise form
+
+           Examples
+           ========
+
+           >>> from sympy import DiracDelta, Piecewise, Symbol, oo
+           >>> x = Symbol('x', real=True)
+
+           >>> DiracDelta(x).rewrite(Piecewise)
+           Piecewise((oo, Eq(x, 0)), (0, True))
+
+           >>> DiracDelta(x - 5).rewrite(Piecewise)
+           Piecewise((oo, Eq(x - 5, 0)), (0, True))
+
+           >>> DiracDelta(x - 5, 4).rewrite(Piecewise)
+           DiracDelta(x - 5, 4)
+
+           >>> y = Symbol('y')
+
+           >>> DiracDelta(y).rewrite(Piecewise)
+           Piecewise((oo, Eq(y, 0)), (0, True))
+
+           >>> DiracDelta(y, 1).rewrite(Piecewise)
+           DiracDelta(y, 1)
+
+           >>> z = Symbol('z', complex=True)
+
+           >>> DiracDelta(z).rewrite(Piecewise)
+           DiracDelta(z)
+
+        """
+        if (self.args[0].is_real or not (self.args[0].is_real is None and self.args[0].is_complex)) and len(args) == 1:
```

Comment:
Why not place these checks in ``eval``? ``len(args)`` will stay here ofcourse.

Something like,

```
if self.args[0].is_real is False:
  <exception>
```

Allow all those that are have ``is_real`` as ``None`` or ``True``.

We can refine them later, if required.